### PR TITLE
NotificationDetails: StackViews

### PR DESCRIPTION
**Pull request by jleandroperez**
#### Details:
This PR addresses technical debt in NotificationDetailsViewController. Specifically:
- UIStackView has been implemented, to easily handle scenarios in which ReplyTextView has to be attached.
- The new leading/trailing Anchor's API has been wired
- Since i was around... cleaned up old Storyboard Segues that weren't used, and shifted over to code for transitions that don't have, as a destination, a VC that lives within Notifications.storyboard.
- Implemented IntrinsicTableView, to simplify "Vertically Centered" layouts, such as the ones used by Badge Notifications

Ref. #5613

--

#### Scenario: Automattcher Segue
1. Log into a dotcom account
2. Open the Notifications List, and press over an automattcher notification

Please, verify that the Reader shows up onscreen

#### Scenario: Comments Segue
1. Open a "Mention in a comment notification"
2. Verify that the Notification Details are onscreen
3. Press over the header

Please, verify that the Comments show up onscreen.

#### Scenario: Comments Segue
1. Open a `New Likes` Notification
2. Press over the header (first row!)

Please, verify that the ReaderPost is onscreen

#### Scenario: Badge Notifications
1. Open a Badge Notification: Stats Surge, First Post, *random* achievement

Please, verify that the interface effectively aligns the UI at the center. The image should show up with an animation.

**Note:** please, check this one on both, iPhone + iPad Devices
**Note II:** please, verify that the verticalScroll gets effectively disabled, in this particular scenario.

#### Scenario: ReplyTextView
1. Open a comment notification (such as a reply)
2. Verify that the ReplyTextView shows up at the bottom of the table
3. Press over the field

Verify that the Reply field gets animated / repositioned. Please, press over the background, and verify that the field gets dismissed.

**Note:** The replyTextField will only be visible on iPhone devices. I'll be iterating over this in other PR's, it should, instead, show up when the horizontal space is compressed!.

-- 

Needs review: @kurzee 
Sir, may i bother you with a review?

Thanks in advance for testing (+ for the patience!!)

### CODE REVIEW

**Comment by kurzee** [comment]
Just double checking:

>Note: The replyTextField will only be visible on iPad devices

Supposed to be:
>Note: The replyTextField will not be visible on iPad devices

@jleandroperez ?

**Comment by jleandroperez** [comment]
@kurzee :facepalm: It's not visible on iPad devices, sorry about the confusion!

**Comment by kurzee** [comment]
On a 3.5 inch iPhone screen, in landscape, (iPhone 4S simulator) the badge notification layout can potentially cut off the top of the badge icon. (Probably a larger issue to fix than the issue itself)

simulator screen shot jul 19 2016 12 09 02 pm

File: WordPress/Classes/ViewRelated/Notifications/Controllers/NotificationDetailsViewController.m [hunk 1]
```diff
@@ -61,13 +54,17 @@
 @interface NotificationDetailsViewController () <ReplyTextViewDelegate, SuggestionsTableViewDelegate>
 
 // Outlets
+@property (nonatomic,   weak) IBOutlet UIStackView          *stackView;
```
Comment (kurzee) [inline WordPress/Classes/ViewRelated/Notifications/Controllers/NotificationDetailsViewController.m]: FWIW, we have been using `strong` for IBOutlets throughout other areas of the app.

**Comment by kurzee** [comment]
Some times the badge notification detail view can cut off the end of the line of text. (I can DM more examples)

simulator screen shot jul 19 2016 12 21 04 pm

**Comment by jleandroperez** [comment]
@kurzee wow! do you have repro steps for that one? or it just happens randomly?

**Comment by kurzee** [comment]
In `NotificationDetailsViewController`:

```
- (void)attachReplyViewIfNeeded
{
    // iPad: We've got a different UI!
    if ([UIDevice isPad]) {
        return;
    }
```
Can we go ahead and change  to `[WPDeviceIdentification isiPad]`?

File: WordPress/Classes/ViewRelated/Notifications/Controllers/NotificationDetailsViewController.m [hunk 2]
```diff
@@ -1293,40 +1258,6 @@ - (void)handleCommentUpdateErrorWithBlock:(NotificationBlock *)block content:(NS
 }
 
 
-#pragma mark - Storyboard Helpers
```
Comment (kurzee) [inline WordPress/Classes/ViewRelated/Notifications/Controllers/NotificationDetailsViewController.m]: dat sweet sweet red

File: WordPress/Classes/ViewRelated/Notifications/Tools/IntrinsicTableView.swift [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+/// will be automatically calculated, based on the Table's Content Size.
+///
+/// Ref.:
+///     -   https://developer.apple.com/library/ios/technotes/tn2154/_index.html#//apple_ref/doc/uid/DTS40013309
+///     -   http://stackoverflow.com/questions/17334478/uitableview-within-uiscrollview-using-autolayout
+///
+class IntrinsicTableView: UITableView
+{
+    override func intrinsicContentSize() -> CGSize {
+        return CGSizeMake(UIViewNoIntrinsicMetric, contentSize.height)
```
Comment (kurzee) [inline WordPress/Classes/ViewRelated/Notifications/Tools/IntrinsicTableView.swift]: I don't suppose we should rather use `super.intrinsicContentSize().height` vs `contentSize.height`?

**Comment by kurzee** [comment]
@jleandroperez all done with this round! Just those few comments, all tests pass. Let me know when ready for round 2!

File: WordPress/Classes/ViewRelated/Notifications/Tools/IntrinsicTableView.swift (hunk 3 above)
Comment (jleandroperez) [inline WordPress/Classes/ViewRelated/Notifications/Tools/IntrinsicTableView.swift]: Nope, the problem is precisely that the intrinsicContentSize isn't being calculated (for TableView's). Thanks for checking!!

**Comment by jleandroperez** [comment]
@kurzee ready for round 2! thanks!!!

File: WordPress/Classes/ViewRelated/Notifications/Tools/IntrinsicTableView.swift [hunk 4]
```diff
@@ -11,7 +11,14 @@ import Foundation
...
 class IntrinsicTableView: UITableView
 {
+    override var contentSize:CGSize {
+        didSet {
+            self.invalidateIntrinsicContentSize()
+        }
+    }
+
     override func intrinsicContentSize() -> CGSize {
+        layoutIfNeeded()
```
Comment (kurzee) [inline WordPress/Classes/ViewRelated/Notifications/Tools/IntrinsicTableView.swift]: Hmmm, not so sure about forcing layout within a calculated layout method. Might lead to issue down the road.

Comment (kurzee) [inline WordPress/Classes/ViewRelated/Notifications/Tools/IntrinsicTableView.swift]: I guess this is OK though as the superview is really the one asking for this size, baed on the view's layout itself.

**Comment by kurzee** [comment]
@jleandroperez all good!! :shipit: 🍧

**Comment by jleandroperez** [comment]
@kurzee thanks a loooooot for reviewing this one!!

**Comment by kurzee** [comment]
It looks like we lost our custom iPad top margin from this merge. Unless we wanted to drop it, @jleandroperez ?

simulator screen shot jul 19 2016 5 11 58 pm

**Pull request merged**